src/rgw/rgw_pubsub.h

```diff
@@ -422,6 +428,13 @@ struct rgw_pubsub_topic {
     if (struct_v >= 2) {
       decode(dest, bl);
       decode(arn, bl);
+
+      // fill in the persistent queue's location if it wasn't present.
+      // rgw_pubsub_dest::decode() couldn't handle this itself because of the
+      // dependency on rgw_pubsub_topic::name
+      if (dest.persistent && dest.persistent_queue.empty()) {
+        dest.persistent_queue = name;
```

Comment:
with this fix, will we have to change something in the v2 migration code, or wil lthat happen automatically?

Reply:
topic migration doesn't delete/recreate these queues, so we shouldn't need to do anything special here. pre-squid topics will use the old queue name, and post-squid topics will use the new namespaced format. v2 topic migration will preserve the queue name either way